tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Stores.AzureAppService/AzureWebsiteStore.cs

```diff
@@ -0,0 +1,114 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+using Azure.ResourceManager;
+using Azure.ResourceManager.AppService;
+using Azure.Sdk.Tools.SecretRotation.Configuration;
+using Azure.Sdk.Tools.SecretRotation.Core;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Sdk.Tools.SecretRotation.Stores.AzureAppService;
+
+public class AzureWebsiteStore : SecretStore
+{
+    public const string MappingKey = "Azure Website";
+
+    private readonly TokenCredential credential;
+    private readonly ILogger logger;
+    private readonly string subscriptionId;
+    private readonly string resourceGroupName;
+    private readonly string websiteName;
+    private readonly RotationAction rotationAction;
+
+    public AzureWebsiteStore(string subscriptionId, string resourceGroupName, string websiteName, RotationAction rotationAction,
+        TokenCredential credential, ILogger logger)
+    {
+        this.subscriptionId = subscriptionId;
+        this.resourceGroupName = resourceGroupName;
+        this.websiteName = websiteName;
+        this.rotationAction = rotationAction;
+        this.credential = credential;
+        this.logger = logger;
+    }
+
+    public override bool CanWrite => true;
+
+    public static Func<StoreConfiguration, SecretStore> GetSecretStoreFactory(TokenCredential credential, ILogger logger)
+    {
+        return configuration =>
+        {
+            var parameters = configuration.Parameters?.Deserialize<Parameters>();
+
+            if (string.IsNullOrEmpty(parameters?.SubscriptionId))
+            {
+                throw new Exception("Missing required parameter 'subscriptionId'");
+            }
+
+            if (string.IsNullOrEmpty(parameters?.ResourceGroup))
+            {
+                throw new Exception("Missing required parameter 'resourceGroup'");
+            }
+
+            if (string.IsNullOrEmpty(parameters?.Website))
+            {
+                throw new Exception("Missing required parameter 'website'");
+            }
+
+            return new AzureWebsiteStore(
+                parameters.SubscriptionId,
+                parameters.ResourceGroup,
+                parameters.Website,
+                parameters.RotationAction,
+                credential,
+                logger);
+        };
+    }
+
+    public override async Task WriteSecretAsync(SecretValue secretValue, SecretState currentState, DateTimeOffset? revokeAfterDate, bool whatIf)
+    {
+        ResourceIdentifier resourceId = WebSiteResource.CreateResourceIdentifier(this.subscriptionId, this.resourceGroupName, this.websiteName);
+        ArmClient client = new ArmClient(this.credential);
+        WebSiteResource website = client.GetWebSiteResource(resourceId);
```

Comment:
Do we want to wrap all of this in the whatIf? Or is whatIf used in the tool to still perform reads.

Reply:
I want whatif to do everything except for volatile writes.   I'd like to catch access and configuration issues in --what-if if possible. i.e.  --what-if shouldn't tell me "WHATIF: Restart Azure Website foo" if the current principal doesn't have access to the subscription.

Reply:
I want the user to feel safe doing a --what-if whenever.   I also want --what-if to find as many configuration and access issues as possible.